File: submissions/KINGINTHENORTH/problem1.cpp
#include<stdio.h>
int fact(int);
int main(){
	int n,x,a,b,c,a1,b1,c1,a2,b2,c2;
	
	scanf("%d",&n);
	a1=n%10;
	a=n/10;
	a2=fact(a1);
	b=a/10;
	b1=a%10;
	b2=fact(b1);
	c=b/10;
	c1=b%10;
	c2=fact(c1);
	if(a2+b2+c2==n)
	{
		printf("Yes");
	}
	else if(a2+b2+c2!=n)
	printf("No");
	return 0;
}
int fact(int a){
if(a==1)
return 1;
else return a*fact(a-1);
}

File: submissions/CodeX/problem1.java
import java.util.*;
public class problem1{
	
	    public static void main(String args[]) {
	        int n = 0, temp = 0, r = 0, s = 0,fact=1;
	        Scanner in = new Scanner(System.in);
	//        System.out.println("Enter a number ");
   n = in.nextInt();
	      //  if (in.hasNextInt()) {
	          // if there is another number  
	        
	        temp = n;
	        while (n != 0) {
	            r = n % 10;
	            
	            for (int i = 1; i <= r; i++) {
					fact = fact * i;
				}
				
	            s = s + fact;
	            n = n / 10;
	            fact=1;
	        }
	        

	        if (temp == s) {
	            System.out.println("Yes");
	        } else {
	            System.out.println("No");
	        }
	    }
} 

